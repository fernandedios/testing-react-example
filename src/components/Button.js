import React from 'react';

const Button = ({ name, handleClick }) => {
  return (
    <button name={name} onClick={handleClick} className="btn btn-primary">{name}</button>
  );
}

export { Button };
