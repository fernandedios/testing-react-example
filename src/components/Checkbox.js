import React, { Component } from 'react';

class Checkbox extends Component {
  constructor(props) {
    super(props);

    this.state = { isChecked: false };
    this.onChange = this.onChange.bind(this);
  }

  onChange() {
    this.setState({ isChecked: !this.state.isChecked });
  }

  render() {
    return (
      <label>
        <input type="checkbox" className="form-control" checked={this.state.isChecked} onChange={this.onChange} />
        {this.state.isChecked ? this.props.labelOn : this.props.labelOff}
      </label>
    );
  }
}

export { Checkbox };
