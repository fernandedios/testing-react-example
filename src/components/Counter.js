import React, { Component } from 'react';

const doIncrement = (prevState) => ({
  counter: prevState.counter + 1
});

const doDecrement = (prevState) => ({
  counter: prevState.counter - 1
});

class Counter extends Component {
  constructor(props) {
    super(props);
    this.state = { counter: 0 };

    this.onIncrement = this.onIncrement.bind(this);
    this.onDecrement = this.onDecrement.bind(this);
  }

  onIncrement() {
    this.setState(doIncrement);
  }

  onDecrement() {
    this.setState(doDecrement);
  }

  render() {
    return (
      <div className="counter">
        <h3>Counter Component</h3>
        <p>{this.state.counter}</p>
        <button onClick={this.onIncrement}>Increment</button>
        <button onClick={this.onDecrement}>Decrement</button>
      </div>
    );
  }
}

export { Counter, doIncrement, doDecrement };
