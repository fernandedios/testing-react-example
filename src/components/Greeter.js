import React from 'react';

const Greeter = ({ name, greetings }) => {
  return <div>{`${greetings} ${name}!`}</div>;
};

Greeter.defaultProps = {
  name: 'John',
  greetings: 'Hello'
};

export { Greeter };
