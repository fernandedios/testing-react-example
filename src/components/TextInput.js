import React from 'react';

const TextInput = ({ type, name, ...others }) => {
  return <input {...others} name={name} id={name} type={type} className="form-control" />;
};

TextInput.defaultProps = { type: 'text' };

export { TextInput };
