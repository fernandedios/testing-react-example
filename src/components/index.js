export * from './Button';
export * from './UsersList';
export * from './Checkbox';
export * from './TextInput';
export * from './LoginForm';
export * from './Greeter';
