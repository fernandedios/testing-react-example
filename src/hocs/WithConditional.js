import React from 'react';

const WithConditional = (Component) => {
  return ({ condition, ...others }) => {
    if (condition) {
      return <Component {...others} />;
    }
    return null;
  }
};

export default WithConditional;
