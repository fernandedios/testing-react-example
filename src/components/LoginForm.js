import React, { Component } from 'react';
import { TextInput } from './TextInput';
import { Button } from './Button';

class LoginForm extends Component {
  constructor(props) {
    super(props);

    this.state = { email: '', password: '' };
    this.onChange = this.onChange.bind(this);
  }

  onChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  render() {
    return (
      <form className="login-form">
        <label>Username: </label>
        <TextInput name="email" onChange={this.onChange} />
        <label>Password: </label>
        <TextInput type="password" name="password" onChange={this.onChange} />
        <Button id="login-button" name="Submit" />
      </form>
    )
  }
}

export { LoginForm };
