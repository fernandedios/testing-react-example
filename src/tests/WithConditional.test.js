import React from 'react';
import { shallow } from 'enzyme';
import ReactTestRenderer from 'react-test-renderer';

import { Greeter } from '../components';
import WithConditional from '../hocs/WithConditional';

describe('<WithConditional />', () => {
  it('should be defined', () => {
    expect(WithConditional).toBeDefined();
  });

  it('should only render if condition evaluates to true', () => {
    const ConditionalComponent = WithConditional(Greeter);
    const output = shallow(<ConditionalComponent condition={true} />);
    expect(output.html()).not.toBe(null);
  });

  it('should NOT render if condition evaluates to false', () => {
    const ConditionalComponent = WithConditional(Greeter);
    const output = shallow(<ConditionalComponent condition={false} />);
    expect(output.html()).toBe(null);
  });

  it('should pass through additional props', () => {
    const name = 'Fernan';
    const greeting = 'Hiya';
    const ConditionalComponent = WithConditional(Greeter);
    const output = shallow(<ConditionalComponent greetings={greeting} name={name} condition={true} />);
    expect(output.html()).toEqual(`<div id="greetings">${greeting} ${name}!</div>`);
  });

  it('should render correctly', () => {
    const ConditionalComponent = WithConditional(Greeter);
    const output = ReactTestRenderer.create(<ConditionalComponent condition={true} />).toJSON();
    expect(output).toMatchSnapshot();
  });
});
