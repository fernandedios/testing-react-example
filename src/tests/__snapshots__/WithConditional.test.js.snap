// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WithConditional /> should render correctly 1`] = `
<div>
  Hello John!
</div>
`;
